receiver/awscontainerinsightreceiver/internal/k8swindows/extractors/extractorhelpers.go

```diff
@@ -1,42 +1,116 @@
 package extractors
 
 import (
+	"fmt"
+
 	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
+// convertCPUStats Convert kubelet CPU stats to Raw CPU stats
+func convertCPUStats(kubeletCPUStat stats.CPUStats) CPUStat {
+	var cpuStat CPUStat
+
+	cpuStat.Time = kubeletCPUStat.Time.Time
+
+	if kubeletCPUStat.UsageCoreNanoSeconds != nil {
+		cpuStat.UsageCoreNanoSeconds = *kubeletCPUStat.UsageCoreNanoSeconds
+	}
+	if kubeletCPUStat.UsageNanoCores != nil {
+		cpuStat.UsageNanoCores = *kubeletCPUStat.UsageNanoCores
+	}
+	return cpuStat
+}
+
+// convertMemoryStats Convert kubelet memory stats to Raw memory stats
+func convertMemoryStats(kubeletMemoryStat stats.MemoryStats) MemoryStat {
+	var memoryStat MemoryStat
+
+	memoryStat.Time = kubeletMemoryStat.Time.Time
+
+	if kubeletMemoryStat.UsageBytes != nil {
+		memoryStat.UsageBytes = *kubeletMemoryStat.UsageBytes
+	}
+	if kubeletMemoryStat.AvailableBytes != nil {
+		memoryStat.AvailableBytes = *kubeletMemoryStat.AvailableBytes
+	}
+	if kubeletMemoryStat.WorkingSetBytes != nil {
+		memoryStat.WorkingSetBytes = *kubeletMemoryStat.WorkingSetBytes
+	}
+	if kubeletMemoryStat.RSSBytes != nil {
+		memoryStat.RSSBytes = *kubeletMemoryStat.RSSBytes
+	}
+	if kubeletMemoryStat.PageFaults != nil {
+		memoryStat.PageFaults = *kubeletMemoryStat.PageFaults
+	}
+	if kubeletMemoryStat.MajorPageFaults != nil {
+		memoryStat.MajorPageFaults = *kubeletMemoryStat.MajorPageFaults
+	}
+	return memoryStat
+}
+
 // ConvertPodToRaw Converts Kubelet Pod stats to RawMetric.
-func ConvertPodToRaw(podStat *stats.PodStats) *RawMetric {
-	var rawMetic *RawMetric
-	rawMetic = &RawMetric{}
+func ConvertPodToRaw(podStat stats.PodStats) RawMetric {
+	var rawMetic RawMetric
+
 	rawMetic.Id = podStat.PodRef.UID
 	rawMetic.Name = podStat.PodRef.Name
 	rawMetic.Namespace = podStat.PodRef.Namespace
 
 	if podStat.CPU != nil {
 		rawMetic.Time = podStat.CPU.Time.Time
```

Comment:
Shouldn't we null check the Time as well?podStat.CPU.Time?

Reply:
Time is not pointer in `podStat.CPU` so it will always have default value or zero value